frontend/src/pages/pipelines/global/experiments/DeleteExperimentModal.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import DeleteModal from '~/pages/projects/components/DeleteModal';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import { ExperimentKFv2 } from '~/concepts/pipelines/kfTypes';
+
+type DeleteExperimentModalProps = {
+  experiment?: ExperimentKFv2;
+  onCancel: () => void;
+};
+
+const DeleteExperimentModal: React.FC<DeleteExperimentModalProps> = ({ experiment, onCancel }) => {
+  const [isDeleting, setDeleting] = React.useState(false);
+  const [error, setError] = React.useState<Error | undefined>();
+  const { api, refreshAllAPI } = usePipelinesAPI();
+
+  if (!experiment) {
+    return null;
+  }
```

Comment:
is this needed? shouldn't the DeleteModal handle this logic

Reply:
If I don't add this type guard, the API call below will complain `await api.deleteExperiment({}, experiment.experiment_id);`